### PR TITLE
fix for redundant partition migration when mancenter is configured

### LINKED ISSUE #11598: Defining ManCenter in config causes PartitonMigration even though there is no data
When ManCenter is enabled, `TimedMemberStateFactory.createMemState` method get stats from Members. `LocalMapStatsProvider.createAllLocalMapStats` method calls `partitionService.getPartition` method without the `triggerOwnerAssignment` boolean flag. In a big cluster with no initial data, this causes partition migration for each joining member & increase cluster startup time significantly.

There may be other sources that causes this, like Diagnostics etc. It'd be good to check all of them if possible.

This is currently affecting a customer with > 100 member cluster.

### PR DESCRIPTION
**Pull request by hasancelik**
fixes #11598

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/map/LocalMapStatsProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        final HazelcastInstance instance = createHazelcastInstance(config);
+
+        assertTrueAllTheTime(new AssertTask() {
+            @Override
+            public void run() throws Exception {
+                //check partition migration triggered or not
+                int partitionStateVersion = getNode(instance).getPartitionService().getPartitionStateVersion();
+                assertEquals(0, partitionStateVersion);
+            }
+        }, 5);
```
Comment (ahmetmircik) [inline hazelcast/src/test/java/com/hazelcast/map/LocalMapStatsProviderTest.java]: 5 seconds can make test fragile, default duration seconds can be used.

**Comment by ahmetmircik** [comment]
@hasancelik is there any other service causing the same defect?

File: hazelcast/src/test/java/com/hazelcast/map/LocalMapStatsProviderTest.java (hunk 1 above)
Comment (hasancelik) [reply hazelcast/src/test/java/com/hazelcast/map/LocalMapStatsProviderTest.java]: I am using *assertTrueAllTheTime* so if i increase *durationSeconds*, test duration will be increased as well.

**Comment by hasancelik** [comment]
@ahmetmircik I have checked service classes which are used by `TimedMemberStateFactory` class.

**Pull request merged**